runtime/fiber.c

```diff
@@ -583,3 +592,14 @@ CAMLprim value caml_drop_continuation (value cont)
   caml_free_stack(stk);
   return Val_unit;
 }
+
+CAMLprim value caml_fiber_id(value unit)
+{
+  struct stack_info* stack = Caml_state->current_stack;
+
+  CAMLassert(unit == Val_unit);
```

Comment:
We don't do that (asserting that unit parameters are equal to `Val_unit`) in other parts of the runtime system.